darts/utils/data/tabularization.py

```diff
@@ -261,18 +272,55 @@ def create_lagged_data(
         tabularization.create_lagged_component_names : return the lagged features names as a list of strings.
 
     """
-    raise_if(
-        is_training and (target_series is None),
-        "Must specify `target_series` if `is_training = True`.",
-    )
+    if is_training and (target_series is None):
+        raise_log(
+            ValueError("Must specify `target_series` if `is_training = True`."),
+            logger=logger,
+        )
 
     # ensure list of TimeSeries format
     target_series = series2seq(target_series)
     past_covariates = series2seq(past_covariates)
     future_covariates = series2seq(future_covariates)
+
+    # get sample weights
+    if isinstance(sample_weight, str):
+        if sample_weight not in SUPPORTED_SAMPLE_WEIGHT:
+            raise_log(
+                ValueError(
+                    f"Invalid `sample_weight` value: {sample_weight}. "
+                    f"If a string, must be one of: {SUPPORTED_SAMPLE_WEIGHT}."
+                ),
+                logger=logger,
+            )
+        if target_series is None:
+            raise_log(
+                ValueError("Must supply `target_series` when using `sample_weight`."),
+                logger=logger,
+            )
+        # create global time weights based on the longest target series
+        max_len = max(len(target_i) for target_i in target_series)
```

Comment:
Since the weights are shared across all the series in case of multiple series training, should the generated weights have a time index index spanning between the most extremes dates across all series instead of just the longest series (since there is no guarantee of index overlap?)

Reply:
As discussed, the (global) weights are computed based on the length of the longest series to be time agnostic (e.g. we can ignore also ignore different frequencies across multiple series). The weights for each series are extracted from the end of those global weights. That means that each series, irrelevant of when in time it was, will weigh the time steps equally based on the index position relative to the end.